src/k4a_ros_device.cpp

```diff
@@ -205,7 +213,17 @@ K4AROSDevice::K4AROSDevice(const NodeHandle &n, const NodeHandle &p) : k4a_devic
     }
     
     // Register our topics
-    rgb_raw_publisher_ = image_transport_.advertise("rgb/image_raw", 1);
+    if (params_.color_format == "jpeg")
+    {
+        // JPEG images are directly published on 'rgb/image_raw/compressed' so that
+        // others can subscribe to 'rgb/image_raw' with compressed_image_transport.
+        // This technique is described in: http://wiki.ros.org/compressed_image_transport#Publishing_compressed_images_directly
+        rgb_jpeg_publisher_ = node_.advertise<CompressedImage>(node_.resolveName("rgb/image_raw") + "/compressed", 1);
```

Comment:
I'm curious: why do you use `node_.resolveName("rgb/image_raw") + "/compressed"` instead of just `"rgb/image_raw/compressed"` ?

Reply:
For the consistency with `image_transport` we probably want to remap `rgb/image_raw` instead of `rgb/image_raw/compressed`.

Resolving `rgb/image_raw/compressed` does not work well, for instance with the following launch file if `color_format=jpeg`.
```
<launch>
  <arg name="color_format"/> 
  <node pkg="azure_kinect_ros_driver" type="node" name="node">
    <param name="color_format" value="$(arg color_format)" />
    <remap from="rgb/image_raw" to="color/image_raw"/>
  </node>
</launch>
```